src/Services/MediaService.php

```diff
@@ -68,4 +71,23 @@ public function first(array|int $where = []): ?Model
     {
         return $this->repository->first($where);
     }
+
+    public function createPreview(string $name): Model
+    {
+        $link = Storage::path($name);
+        $name = "preview_{$name}";
+        $dirname = pathinfo($link)['dirname'];
+        $newLink = "{$dirname}/{$name}";
+
+        Image::load($link)
+            ->width(config('media.preview.width'))
+            ->height(config('media.preview.height'))
+            ->save($newLink);
```

Comment:
should it store preview in the same storage? It looks strange that we pass link here

Reply:
Fixed.